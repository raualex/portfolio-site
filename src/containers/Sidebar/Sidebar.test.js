import React from 'react';
import { Sidebar, mapStateToProps, mapDispatchToProps } from './';
import { shallow } from 'enzyme';

describe('Sidebar', () => {
  let wrapper;
  let mockFunc = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Sidebar setSelectedNav={mockFunc} />)
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  });

  describe('setNav function', () => {
    it('should fire setSelectedNav with the correct params', () => {
      let mockEvent = { target: { innerHTML: 'Bio' } }
      wrapper.instance().setNav(mockEvent)
      expect(mockFunc).toHaveBeenCalledWith('Bio')
    });
  });
});

describe('mapStateToProps function', () => {
  
  it('should return object with selected Navigation item', () => {
    const mockState = {
      selectedNav: 'Bio',
      somethingElse: 'Not Bio'
    }
    const expected = {
      selectedNav: 'Bio'
    }
    const mappedProps = mapStateToProps(mockState)
    expect(mappedProps).toEqual(expected)
  });
});

describe('mapDispatchToProps function', () => {
  const mockDispatch = jest.fn()
  let mappedProps;

  beforeEach(() => {
    mappedProps = mapDispatchToProps(mockDispatch)
  });

  it('should call dispatch when setSelectedNav is called', () => {
    mappedProps.setSelectedNav()
    expect(mockDispatch).toHaveBeenCalled()
  });
});