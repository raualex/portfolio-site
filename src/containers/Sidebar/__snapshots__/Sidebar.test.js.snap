// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Sidebar
    setSelectedNav={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="Profile Picture"
          className="profile-pic"
          src="profile.jpg"
        />,
        <div
          className="nav-links"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item inactive-item"
            onClick={[Function]}
            to="/bio"
          >
            Bio
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item inactive-item"
            onClick={[Function]}
            to="/skills"
          >
            Skills
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item inactive-item"
            onClick={[Function]}
            to="/projects"
          >
            Projects
          </NavLink>
          <NavLink
            activeClassName="active"
            aria-current="page"
            className="nav-item inactive-item"
            onClick={[Function]}
            to="/contact"
          >
            Contact
          </NavLink>
        </div>,
      ],
      "className": "nav-bar-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "Profile Picture",
          "className": "profile-pic",
          "src": "profile.jpg",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/bio"
            >
              Bio
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/skills"
            >
              Skills
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/projects"
            >
              Projects
            </NavLink>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/contact"
            >
              Contact
            </NavLink>,
          ],
          "className": "nav-links",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Bio",
              "className": "nav-item inactive-item",
              "onClick": [Function],
              "to": "/bio",
            },
            "ref": null,
            "rendered": "Bio",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Skills",
              "className": "nav-item inactive-item",
              "onClick": [Function],
              "to": "/skills",
            },
            "ref": null,
            "rendered": "Skills",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Projects",
              "className": "nav-item inactive-item",
              "onClick": [Function],
              "to": "/projects",
            },
            "ref": null,
            "rendered": "Projects",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Contact",
              "className": "nav-item inactive-item",
              "onClick": [Function],
              "to": "/contact",
            },
            "ref": null,
            "rendered": "Contact",
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="Profile Picture"
            className="profile-pic"
            src="profile.jpg"
          />,
          <div
            className="nav-links"
          >
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/bio"
            >
              Bio
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/skills"
            >
              Skills
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/projects"
            >
              Projects
            </NavLink>
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="nav-item inactive-item"
              onClick={[Function]}
              to="/contact"
            >
              Contact
            </NavLink>
          </div>,
        ],
        "className": "nav-bar-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "Profile Picture",
            "className": "profile-pic",
            "src": "profile.jpg",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-item inactive-item"
                onClick={[Function]}
                to="/bio"
              >
                Bio
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-item inactive-item"
                onClick={[Function]}
                to="/skills"
              >
                Skills
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-item inactive-item"
                onClick={[Function]}
                to="/projects"
              >
                Projects
              </NavLink>,
              <NavLink
                activeClassName="active"
                aria-current="page"
                className="nav-item inactive-item"
                onClick={[Function]}
                to="/contact"
              >
                Contact
              </NavLink>,
            ],
            "className": "nav-links",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Bio",
                "className": "nav-item inactive-item",
                "onClick": [Function],
                "to": "/bio",
              },
              "ref": null,
              "rendered": "Bio",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Skills",
                "className": "nav-item inactive-item",
                "onClick": [Function],
                "to": "/skills",
              },
              "ref": null,
              "rendered": "Skills",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Projects",
                "className": "nav-item inactive-item",
                "onClick": [Function],
                "to": "/projects",
              },
              "ref": null,
              "rendered": "Projects",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Contact",
                "className": "nav-item inactive-item",
                "onClick": [Function],
                "to": "/contact",
              },
              "ref": null,
              "rendered": "Contact",
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
